dashboard/views.py

```diff
@@ -128,6 +129,17 @@ def get_course_info(request, course_id=0):
     resp['course_view_options'] = CourseViewOption.objects.get(course=course).json(include_id=False)
     resp['resource_types'] = course_resource_list
 
+    try:
+        course_users = User.objects.get(course_id=course_id)
+        resp['course_user_exist'] = 1
+    except ObjectDoesNotExist:
+        logger.info("no user for this course")
+        resp['course_user_exist'] = 0
+    except MultipleObjectsReturned:
+        # more than one user exist
+        resp['course_user_exist'] = 1
```

Comment:
Hm, I think leveraging the `User` table is OK, although if we ever get around to changing this to `Enrollment`, it's one more thing that needs to be updated. It would be nice if we could use `CronJobLog` (which could be a more direct way of knowing if a data load happened), but then we'd also need some way to determine when the course was added, which we don't really right now...

Rather than using `try/except` here, could you just use a `QuerySet`, like `User.objects.filter(course_id=course_id)`? Then you could call `len()` on that, and if it's more than `0`, set `resp['course_user_exists']`. Same result, maybe a little less code, at least one less import.

Reply:
Fixed.